sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/directconnectivity/ReflectionUtils.java

```diff
@@ -186,6 +186,11 @@ public static void setDiagnosticsProvider(CosmosAsyncClient cosmosAsyncClient, D
 
     public static void setClientTelemetryConfig(CosmosAsyncClient cosmosAsyncClient, CosmosClientTelemetryConfig cfg){
         set(cosmosAsyncClient, cfg, "clientTelemetryConfig");
+        AsyncDocumentClient asyncClient = get(
+            AsyncDocumentClient.class, cosmosAsyncClient, "asyncDocumentClient");
+        if (asyncClient instanceof RxDocumentClientImpl) {
+            set(((RxDocumentClientImpl)asyncClient), cfg, "clientTelemetryConfig");
```

Comment:
Can this explicit cast be removed since you're passing a subtype of `AsyncDocumentClient`?

Reply:
No - AsyncDocumentClient doesn't have the private field.